Sources/ContainerClient/Utility.swift

```diff
@@ -305,4 +277,47 @@ public struct Utility {
         }
         return result
     }
+
+    /// Gets an existing volume or creates it if it doesn't exist.
+    /// Shows a warning for named volumes when auto-creating.
+    private static func getOrCreateVolume(parsed: ParsedVolume) async throws -> Volume {
+        let labels = parsed.isAnonymous ? [Volume.anonymousLabel: ""] : [:]
+
+        let volume: Volume
+        let volumeCreated: Bool
+        do {
+            volume = try await ClientVolume.create(
+                name: parsed.name,
+                driver: "local",
+                driverOpts: [:],
+                labels: labels
+            )
+            volumeCreated = true
+        } catch let error as VolumeError {
+            guard case .volumeAlreadyExists = error else {
+                throw error
+            }
+            // Volume already exists, just inspect it
+            volume = try await ClientVolume.inspect(parsed.name)
+            volumeCreated = false
+        } catch let error as ContainerizationError {
+            // Handle XPC-wrapped volumeAlreadyExists error
+            guard error.message.contains("already exists") else {
+                throw error
+            }
+            volume = try await ClientVolume.inspect(parsed.name)
+            volumeCreated = false
+        }
+
+        // Warn user if named volume was auto-created
+        if volumeCreated && !parsed.isAnonymous {
+            let warning = "Volume '\(parsed.name)' does not exist. Creating new volume."
+            let formattedWarning = "\u{001B}[33m\(warning)\u{001B}[0m\n"
```

Comment:
We're not logging anything else in utility today, let's drop the output here for now and add a TODO comment mentioning that we might want to warn the user in the future.

We have #642 for improving the way we report things. At some point we should consider plumbing ContainerClient for Logger so we can provide more, and consistent feedback, using the log handler or a facility atop it (similar to  [this](https://github.com/Textualize/rich)) to do things like ANSI vs plain, etc.